src/autocomplete.js

```diff
@@ -231,8 +231,15 @@ class Autocomplete {
             }
         }
 
+        // posGhostText can be below the editor rendering the popup away from the editor.
+        // In this case, we want to render the popup such that the top aligns with the bottom of the editor.
+        var editorContainerBottom = editor.container.getBoundingClientRect().bottom - lineHeight;
+        var lowestPosition = editorContainerBottom < posGhostText.top ?
+            {top: editorContainerBottom, left: posGhostText.left} :
```

Comment:
> "Added line #L238 was not covered by tests"

But you have a test that tests these changes, in particular it tests whether "Popup should start one pixel below the bottom of the editor". 

Do you think it would be valuable to also test the other edge case in your test?

Reply:
you mean the edge case where the popup is within the editor and should therefore be rendered at the bottom of the ghost text?

Reply:
added an extra test to the PR